.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,58 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Publish Docker image
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  push_to_registry:
+    name: Push Docker image to Docker Hub
+    if: ${{ startsWith(github.repository, 'apache/') }}
+    runs-on: ubuntu-22.04
+    strategy:
+       matrix:
+         include:
+           - suffix-name: ''
+             opts: '--flink-provided --spark-provided --hive-provided'
+           - suffix-name: '-spark'
+             opts: '--flink-provided --hive-provided'
+           - suffix-name: '-flink'
+             opts: '--spark-provided --hive-provided'
+           - suffix-name: '-all'
+             opts: ''
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USER }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build Kyuubi Docker Image
+        run: |
+          wget -O kyuubi.tgz https://archive.apache.org/dist/kyuubi/kyuubi-$(cat release/release_version)/apache-kyuubi-$(cat release/release_version)-source.tgz
+          tar -xvf kyuubi.tgz --strip-components=1
+          ./build/dist ${{ matrix.opts }} -Dmaven.javadoc.skip=true -Dmaven.scaladoc.skip=true -Drat.skip=true -Dscalastyle.skip=true -Dspotless.check.skip -DskipTests
+          docker build --tag apache/kyuubi:$(cat release/release_version)${{ matrix.suffix-name }} --file $(cat release/release_version)/*/Dockerfile .
+      - name: Validate docker image
+        run: docker images
+      - name: Push Docker image
+        run: docker push apache/kyuubi:$(cat release/release_version)${{ matrix.suffix-name }}
```

Comment:
Why not use `docker/build-push-action` action for pushing images?

Reply:
using command is easy for testing at local.. does that action contains some feautre ?

Reply:
With `docker/build-push-action`, potential benefits are,
- `kyuubi` is using this action in workflows
- use actions in config definition rather than command line code
- this action should have cover all the expected feature we want
- maybe better integration with `docker/setup-buildx-action` and `docker/setup-qemu-action`, as they are mentioned and recommended in docs of `docker/build-push-action`

For local build testing, we could provide another script for it, and it will be handy to run and modify locally.

Reply:
not see the benefits... separate command to script and use github action just introduce complexity. We can improve the workflow in future if necessary